Automattic/jetpack
# Add the export modal for the new Jetpack Forms dashboard

## Proposed changes:

This PR adds an export modal for the new Jetpack Forms dashboard. The modal should work similarly to the existing one on the legacy view.

In addition to the modal, I've also updated the export handler itself so it's now also compatible with post status and search filters. This way, the export should always contain the same results you're seeing in the dashboard unless you make a specific selection.

_Note: The selection mechanism is a little buggy. I've moved it to the state layer but I saw @jcheringer also made some changes in #29766. Let's merge that one first, and then I'll fix any remaining issues here._

<img width="791" alt="Screenshot 2023-03-29 at 13 42 00" src="https://user-images.githubusercontent.com/8056203/228526274-5b3c84f5-5615-4ceb-bc64-786deddfc4d8.png">

<img width="781" alt="Screenshot 2023-03-29 at 13 42 28" src="https://user-images.githubusercontent.com/8056203/228526343-751039b3-6f91-4c13-9595-86b1e800481e.png">

### Other information:


## Jetpack product discussion



pcsBup-Pl-p2

## Does this pull request change what data or activity we track or use?



No.

## Testing instructions:






- Go to `admin.php?page=jetpack-forms`.
- There should be an 'Export' button in the upper right corner above the table.
- Clicking the button should display the export modal.
- Clicking 'Download' under the CSV section should trigger a download for the CSV file.
- The Google Drive section should have an export button when you're already connected to google drive, otherwise a connect button will be shown.
- Triggering the google drive export should open a new tab with the newly created export document.
- In both cases, the export should contain the same responses you're currently seeing inside the inbox.
- If you change the selection, only the selected elements should be exported.
